zendesk/resource_zendesk_trigger_test.go

```diff
@@ -1,22 +1,201 @@
+	}
+
+	if v := i.Get("title"); v != "trigger" {
+		t.Fatalf("CreateTrigger did not set resource title. title was %s", v)
+	}
+}
+
+func TestReadTrigger(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := newIdentifiableGetterSetter()
+	i.SetId("12345")
+
+	expected := zendesk.Trigger{
+		Title:  "trigger",
+		Active: true,
+	}
+	m.EXPECT().GetTrigger(gomock.Any(), gomock.Eq(int64(12345))).Return(expected, nil)
+	if err := readTrigger(i, m); err != nil {
+		t.Fatalf("GetTrigger received an error when calling: %v", err)
+	}
+}
+
+func TestUpdateTrigger(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id:              "12345",
+		mapGetterSetter: mapGetterSetter{},
+	}
+
+	m.EXPECT().UpdateTrigger(gomock.Any(), gomock.Eq(int64(12345)), gomock.Any()).Return(zendesk.Trigger{}, nil)
+	if err := updateTrigger(i, m); err != nil {
+		t.Fatalf("updateTrigger returned an error %v", err)
+	}
+}
+
+func TestDeleteTrigger(t *testing.T) {
 	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
 	c := mock.NewClient(ctrl)
 	d := newIdentifiableGetterSetter()
 
 	d.SetId("1234")
 
-	c.EXPECT().DeleteTrigger(gomock.Eq(int64(1234))).Return(nil)
-	err := resourceZendeskTriggerDelete(d, c)
+	c.EXPECT().DeleteTrigger(gomock.Any(), gomock.Eq(int64(1234))).Return(nil)
+	err := deleteTrigger(d, c)
 	if err != nil {
 		t.Fatalf("Got error from resource delete: %v", err)
 	}
 }
+
+func testTriggerDestroyed(s *terraform.State) error {
+	client := testAccProvider.Meta().(zendesk.TriggerAPI)
+
+	for k, r := range s.RootModule().Resources {
+		if r.Type != "zendesk_trigger" {
+			continue
+		}
+
+		id, err := atoi64(r.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		ctx := context.Background()
+		_, err = client.GetTrigger(ctx, id)
+		if err == nil {
+			return fmt.Errorf("did not get error from zendesk when trying to fetch the destroyed trigger. resource name %s", k)
+		}
+
+		zdresp, ok := err.(zendesk.Error)
+		if !ok {
+			return fmt.Errorf("error %v cannot be asserted as a zendesk error", err)
+		}
+
+		if zdresp.Status() != http.StatusNotFound {
+			return fmt.Errorf("did not get a not found error after destroy. error was %v", zdresp)
+		}
+	}
+	return nil
+}
+
+func TestAccTriggerExample(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testTriggerDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: readExampleConfig(t, "triggers.tf"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("zendesk_trigger.auto-reply-trigger", "title", "Auto Reply Trigger"),
+					resource.TestCheckResourceAttr("zendesk_trigger.auto-reply-trigger", "active", "true"),
+					resource.TestCheckResourceAttrSet("zendesk_trigger.auto-reply-trigger", "all"),
```

Comment:
I attached a debugger and it looks like for these sets the state variables are stored a little different.

You can probably change to this
```suggestion
					resource.TestCheckResourceAttrSet("zendesk_trigger.auto-reply-trigger", "all.#"),
```